src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -334,7 +334,7 @@ private async Task ProcessDocumentAsync(ImmutableArray<IIncrementalAnalyzer> ana
                         {
                             using (Logger.LogBlock(FunctionId.WorkCoordinator_ProcessDocumentAsync, w => w.ToString(), workItem, cancellationToken))
                             {
-                                var textDocument = solution.GetTextDocument(documentId);
+                                var textDocument = solution.GetTextDocument(documentId) ?? await solution.GetSourceGeneratedDocumentAsync(documentId, cancellationToken).ConfigureAwait(false);
```

Comment:
I would ideally expect `GetTextDocument` to itself handle the source generated document case instead of having each caller require the code pattern being added here. That is the core purpose of this extension method, so the callers don't have to deal with each document kind. Is the primary thing blocking that the fact that `GetTextDocument` is sync while `GetSourceGeneratedDocumentAsync` is async, and changing `GetTextDocument` to `GetTextDocumentAsync` would be a much broader change? If so, can we open a tracking issue to clean this up with a follow-up change?

Reply:
> Is the primary thing blocking that the fact that GetTextDocument is sync while GetSourceGeneratedDocumentAsync is async, and changing GetTextDocument to GetTextDocumentAsync would be a much broader change?

Unfortunately, yes. If Project.GetDocument() was async in the first place, we probably wouldn't have needed the other methods.